kv/kvserver/allocator/allocatorimpl: TestAllocatorFullDisks failed

kv/kvserver/allocator/allocatorimpl.TestAllocatorFullDisks [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9332082?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9332082?buildTab=artifacts#/) on release-23.1 @ [aec78f33d45a8376a0ecec885688bae60dbfb85c](https://github.com/cockroachdb/cockroach/commits/aec78f33d45a8376a0ecec885688bae60dbfb85c):


```
=== RUN   TestAllocatorFullDisks
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/5445000f6d8e70a16462d5297e3e2235/logTestAllocatorFullDisks162706032
    test_log_scope.go:79: use -show-logs to present logs inline
    allocator_test.go:8511: testStore 11 ran out of space during generation 60 (rangesAdded=1199/1200): disk (capacity=1.0 GiB, available=-16 MiB, used=1.0 GiB, logicalBytes=1.0 GiB), ranges=65, leases=0, queries=0.00, writes=0.00, ioThreshold={{0 0 0 0}} bytesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
    allocator_test.go:8573: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5445000f6d8e70a16462d5297e3e2235/logTestAllocatorFullDisks162706032
--- FAIL: TestAllocatorFullDisks (2.00s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAllocatorFullDisks.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-26187

This test failed after 20 minutes of stressing on master. I'm going to look further into it now.

```
=== RUN   TestAllocatorFullDisks
    test_log_scope.go:161: test logs captured to: /home/kvoli/go/src/github.com/cockroachdb/testoutput/_tmp/5445000f6d8e70a16462d5297e3e2235/logTestAllocatorFullDisks342762789
    test_log_scope.go:79: use -show-logs to present logs inline
    allocator_test.go:8511: testStore 11 ran out of space during generation 58 (rangesAdded=1197/1200): disk (capacity=1.0 GiB, available=-16 MiB, used=1.0 GiB, logicalBytes=1.0 GiB), ranges=65, leases=0, queries=0.00, writes=0.00, ioThreshold={{0 0 0 0}} bytesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
    allocator_test.go:8573: -- test log scope end --
test logs left over in: /home/kvoli/go/src/github.com/cockroachdb/testoutput/_tmp/5445000f6d8e70a16462d5297e3e2235/logTestAllocatorFullDisks342762789
--- FAIL: TestAllocatorFullDisks (4.14s)
FAIL
I230403 21:36:38.434111 1 (gostd) testmain.go:240  [-] 1  Test //pkg/kv/kvserver/allocator/allocatorimpl:allocatorimpl_test exited with error code 1


ERROR: exit status 1

7835 runs completed, 1 failures, over 20m2s
context canceled

```

--- a/pkg/kv/kvserver/allocator/allocatorimpl/allocator_test.go
+++ b/pkg/kv/kvserver/allocator/allocatorimpl/allocator_test.go
@@ -8475,7 +8475,14 @@ func TestAllocatorFullDisks(t *testing.T) {
 
 	do := makeDiskCapacityOptions(&st.SV)
 
-	rangesPerNode := int(math.Floor(capacity * do.ShedAndBlockAllThreshold / rangeSize))
+	// Each range is equally sized (16mb), we want the number of ranges per node,
+	// when their size is added, to be no greater than the full disk rebalance
+	// threshold (0.925%) e.g for below:
+	//   capacity  = 1024mb
+	//   rangeSize = 16mb
+	//   threshold = 0.925
+	//   rangesPerNode =  ⌊1024mb * 0.925 / 16mb⌋ = 59
+	rangesPerNode := int(math.Floor(capacity * do.RebalanceToThreshold / rangeSize))
 	rangesToAdd := rangesPerNode * nodes
 
 	// Initialize testStores.

